processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -101,7 +101,7 @@
     @Override
     public String apply(final Object o)
     {
-      return String.valueOf(o);
+      return o == null ? null : String.valueOf(o);
```

Comment:
I think it's fine, null/empty dim values will end up as "" via a nullToEmpty() call when added to the dictionary:

https://github.com/navis/druid/blob/invalid-encode-nullstring/processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java#L1125